### DISCUSSION TITLE
Working for months. Now getting cannot connect to MQTT broker

### DISCUSSION THREAD

**Discussion by donnievawter** [poblabs/weewx-belchertown#779] [Q&A]
I have an external MQTT broker. Nothing has changed. I can connect to the broker  with MQTT explorer and see the messages being published. However on the weather page I am getting the following error:
        
       WebSocket connection to 'wss://weather2.sangrephotography.com:9001/mqtt' failed: 
d._doConnect @ paho-mqtt.min.js:37
It had been working for months and I have not changed anything. Updating weewx to 4.8 did not help.
I am user realease 1.2 of Belchertown
The error occurs in both Safari and Edge
Any ideas?

**Comment by ThePeePs** [comment]
What are you using for your MQTT broker?

**Reply by donnievawter** [reply]
mosquitto. Installed on a digital ocean ubuntu droplet. From mqtt explorer I can subscribe to the weather messages and see them fine.   Also I can subscribe to weather/loop from other applications and see the data coming in so it is something specific to the browser based javascript websockets I think. So publishing works fine and subscribing from other applications works fine. subscribing from weather page fails.

**Reply by ThePeePs** [reply]
look and see what version it's running.  I was running something similar, but it was happening with Firefox, instead of Safari and Edge, where the websocket call was in HTTP/2, and mosquitto was [having a hard](https://github.com/poblabs/weewx-belchertown/discussions/777) time with that _and_ TLS, and I was getting a connection error as well.

**Reply by donnievawter** [reply]
I updated to latest version.  For some reason it is getting a permission denied trying to open cert file but even if I disable TLS it still fails on websocket call. I don't know if you can call without TLS if you are running website under https but the error is it can't connect to the broker. I am going to try to set up a web-socket call in node-red to see if it can connect. Will update

**Reply by ThePeePs** [reply]
make sure the user that mosquitto is running under has permissions for the cert file(s) and key(s), (it needs read access to all of them).  you can have a non-tls websocket for a tls page, but it isn't recommended, and most browsers will complain, or not load the "clear-text" content, unless you change browser settings.

**Reply by donnievawter** [reply]
That was it. group on certs files had read access but the group was root. changed the group to mosquito and now it works. I am guess it stopped working when the certs were renewed.  At least I wound up with a newer version of mosquito in the process of troubleshooting. Thanks for helping.

**Reply by ThePeePs** [reply]
welcome, glad i could help.  perms on certs have gotten me, and i'm sure many others, more then once.
